java/common/Loader.java

```diff
@@ -75,15 +74,24 @@ private static Path getNativeResourceURI() throws IOException {
         }
         String platformString = DRIVER_JNI_JAR_NAME.get(platform);
         ClassLoader loader = Loader.class.getClassLoader();
-        URL jniURL = null;
         Iterator<URL> resourceIterator = loader.getResources(DRIVER_JNI_LIBRARY_NAME).asIterator();
+
+        URL jniURL = null;
+        boolean isValid = true;
```

Comment:
I think this will be easier to read/understand quickly if we were verbose like this:
```
URL firstJNI = null;
URL validatedJNI = null;

while (...) {
  resource = ;
  if (firstJNI == null) firstJNI = resource;

  if (...)
    validatedJNI = resource;

 }
 // switch on the final choice of URL here
 ```